Update for compatibility with Kedro > 0.19

DimedS opened the pull request:
This PR ensures compatibility of `kedro-sagemaker` with Kedro versions > 0.19.

## Key Updates
- **Requirements**: Unpinned to support broader compatibility.
- **Dataset**: Updated `DataSet` to `Dataset` to reflect changes in Kedro.
- **Configuration Loader**: Replaced with `OmegaConfLoader` for handling `sagemaker.yaml`.
- **Requirements File**: Moved `/src/requirements.txt` to `/requirements.txt`.
- **Documentation**: Updated to align with the above changes.
- **Tests**: Modified to reflect the updated functionality.

ElenaKhaustova commented on tests/test_generator.py:
```diff
@@ -394,50 +394,50 @@ def test_should_generate_training_steps_and_skip_model_registration(
-        json.loads(estimator.environment["KEDRO_SAGEMAKER_RUNNER_CONFIG"])["bucket"]
-        == "__bucket_name__"
-    )
+# @patch("kedro.framework.project.pipelines", {"__default__": sample_pipeline})
```
Shall we make a note about why these tests are commented?

ElenaKhaustova commented on tests/test_runner.py:
```diff
@@ -26,19 +26,19 @@ def test_can_invoke_dummy_pipeline(
-            catalog,
-        )
-    assert results["output_data"] == input_data, "Invalid output data"
+# def test_runner_fills_missing_datasets(
```
Shall we make a note about why these tests are commented?

DimedS replied:
fixed

ElenaKhaustova reviewed (commented):
Thank you, @DimedS, great work!

I tested all the steps but loading docker image to ECR because of Platform McK issues. So I loaded the workflow manually with the UI.

My main feedback is to extend `Quickstart` section with the overview of what's gonna be done there.

DimedS commented on tests/test_cli.py:
```diff
@@ -75,30 +74,35 @@ def test_can_initialize_basic_plugin_config(
-        assert isinstance(json.loads(output_path.read_text()), dict)
-
-
+# @pytest.mark.parametrize(
```
fixed

DimedS commented on docs/source/03_quickstart.rst:
```diff
@@ -59,10 +50,7 @@ The ``init`` command automatically will create:
-   whereas the plugin will automatically use S3. Only
-   input data is required to be read locally. Final
-   ``conf/base/catalog.yml`` should look like this:
+8. Adjust the Data Catalog. By default, all data is stored locally, but the plugin automatically uses S3 for storage. Only input data needs to be read locally. If you have intermediate data stored in memory, you must specify it in the Data Catalog to save it in persistent storage (S3), as each node will execute in a separate container. The final version of ``conf/base/catalog.yml`` should look li ...
```
Thanks, @ElenaKhaustova! I rewrote it to make it clearer.

DimedS commented:
> Thank you, @DimedS, great work!
> 
> I tested all the steps except loading the Docker image to ECR because of Platform McK issues. So, I loaded the workflow manually with the UI. Everything worked as expected.
> 
> My main feedback is to extend `Quickstart` section with the overview of what's gonna be done there. We can also provide references to the requirements needed to set up on the AWS side.

Many thanks for your review, @ElenaKhaustova! I’ve fixed all the tests, updated the autosaving functionality for MemoryDataset to S3, and revised the documentation based on your comments. Requesting a re-review.

ElenaKhaustova reviewed (approved):
Great work, thank you @DimedS!

em-pe commented:
@DimedS Please add a one sentence description of the change in CHANGELOG.md under Unreleased section.
